BacktraceScreen.c

```diff
@@ -0,0 +1,384 @@
+
+   int maxFunctionNameLength = printingHelper->maxFuncNameLen;
+   if (!!(displayOptions & DEMANGLE_NAME_FUNCTION) &&
+         printingHelper->maxDemangledFuncNameLen > 0) {
+      maxFunctionNameLength = printingHelper->maxDemangledFuncNameLen;
+   }
+
+   char* line = NULL;
+   int len = xAsprintf(&line, "%-*s %-*s %-*s %-*s",
+      (int)printingHelper->maxFrameNumLen, BACTRACE_PANEL_HEADER_NUMBER_FRAME,
+      (int)printingHelper->maxAddrLen, BACTRACE_PANEL_HEADER_ADDRESS,
+      (int)maxFunctionNameLength, BACTRACE_PANEL_HEADER_NAME,
+      (int)printingHelper->maxObjPathLen, BACTRACE_PANEL_HEADER_PATH
+   );
+
+   RichString_appendnAscii(out, CRT_colors[BACKTRACE_HEADER], line, len);
+   free(line);
+}
+
+static void BacktraceFrame_highlightBasename(const BacktraceFrame* frame, RichString* out, char* line, int objectPathStart) {
+   assert(objectPathStart >= 0);
+
+   char* basename = frame->backtracePanel->process->procExe + frame->backtracePanel->process->procExeBasenameOffset;
+   char* object = line + objectPathStart;
+   if (!basename || !object) {
+      return;
+   }
+
+   char* lastSlash = strrchr(object, '/');
+   if (lastSlash) {
+      object = lastSlash + 1;
+   }
+
+   char* objectTrimmed = String_trim(object);
+   if (String_eq(basename, objectTrimmed)) {
+      RichString_setAttrn(out, CRT_colors[PROCESS_BASENAME], object - line, strlen(objectTrimmed));
+   }
+
+   free(objectTrimmed);
+}
+
+static void BacktraceFrame_display(const Object* super, RichString* out) {
+   const BacktraceFrame* frame = (const BacktraceFrame* )super;
+   assert(frame);
+
+   if (frame->index == BACKTRACE_FRAME_HEADER_INDEX) {
+      BacktraceFrame_displayHeader(frame, out);
+      return;
+   }
+
+   const BacktracePanelPrintingHelper* printingHelper = &frame->backtracePanel->printingHelper;
+   assert(printingHelper);
+
+   const BacktraceScreenDisplayOptions displayOptions = frame->backtracePanel->displayOptions;
+
+   char* functionName = frame->functionName;
+   int maxFunctionNameLength = printingHelper->maxFuncNameLen;
+   if (!!(displayOptions & DEMANGLE_NAME_FUNCTION) &&
+         printingHelper->maxDemangledFuncNameLen > 0) {
+      maxFunctionNameLength = printingHelper->maxDemangledFuncNameLen;
+      if (frame->demangleFunctionName) {
+         functionName = frame->demangleFunctionName;
+      }
+   }
+
+   char *completeFunctionName = NULL;
+   xAsprintf(&completeFunctionName, "%s+0x%zx", functionName, frame->offset);
+
+   char* objectDisplayed = frame->objectName;
+   size_t objectLength = printingHelper->maxObjNameLen;
+   if (!!(displayOptions & SHOW_FULL_PATH_OBJECT)) {
+      objectDisplayed = frame->objectPath;
+      objectLength = printingHelper->maxObjPathLen;
+   }
+
+   char* line = NULL;
+   int objectPathStart = -1;
+   int len = xAsprintf(&line, "%-*d 0x%0*zx %-*s %n%-*s",
+      (int)printingHelper->maxFrameNumLen, frame->index,
```

Comment:
How about right aligning the frame number field?

Reply:
Only the right alignment, I'm not a fan. I prefer to pad with 0. Here with all alignment : 
1. Right alignment : 
![image](https://github.com/user-attachments/assets/3c48b696-6e9f-42b9-b20b-f7018a70c0d3)
2. Pad with 0: 
![image](https://github.com/user-attachments/assets/ccf0c415-99fa-4c22-9e0c-9d80b6697a7f)
3. Left alignment : 
![image](https://github.com/user-attachments/assets/05a718f3-467c-422a-b0fb-fabfacaf19f8)

I prefer the 3 (the left alignment), it's aligned with the header.

Reply:
You know the header can be right-aligned, don't you?

Reply:
Go for the 1 (right alignment) :smile: